spec/cucumber/core/test/filters/locations_filter_spec.rb

```diff
@@ -120,20 +120,93 @@ def test_case_named(name)
           expect(receiver.test_case_locations).to eq [test_case_named('two').location]
         end
 
-        it "doesn't match a location after the scenario line" do
+        it "matches the location of a step of the scenario" do
           location = Test::Location.new(file, 9)
           filter = Test::LocationsFilter.new([location])
           compile [doc], receiver, [filter]
-          expect(receiver.test_case_locations).to eq []
+          expect(receiver.test_case_locations).to eq [test_case_named('two').location]
+        end
+
+        it 'matches a location on the last step of the scenario' do
+          location = Test::Location.new(file, 10)
+          filter = Test::LocationsFilter.new([location])
+          compile [doc], receiver, [filter]
+          expect(receiver.test_case_locations).to eq [test_case_named('two').location]
+        end
+
+        xit "matches a location on the scenario's comment" do
+          location = Test::Location.new(file, 6)
+          filter = Test::LocationsFilter.new([location])
+          compile [doc], receiver, [filter]
+          expect(receiver.test_case_locations).to eq [test_case_named('two').location]
         end
 
-        it "doesn't match a location before the scenario line" do
+        it "matches a location on the scenario's tags" do
           location = Test::Location.new(file, 7)
           filter = Test::LocationsFilter.new([location])
           compile [doc], receiver, [filter]
+          expect(receiver.test_case_locations).to eq [test_case_named('two').location]
+        end
+
+        it "doesn't match a location after the last step of the scenario" do
+          location = Test::Location.new(file, 11)
+          filter = Test::LocationsFilter.new([location])
+          compile [doc], receiver, [filter]
           expect(receiver.test_case_locations).to eq []
         end
 
+        it "doesn't match a location before the scenario" do
```

Comment:
This scenario is technically a duplicate of the one above for the reason mentioned.